examples/step-12/step-12.cc

```diff
@@ -548,8 +548,7 @@ namespace Step12
     std::ofstream output(filename);
 
     DataOut<dim> data_out;
-    data_out.attach_dof_handler(dof_handler);
-    data_out.add_data_vector(solution, "u");
+    data_out.add_data_vector(dof_handler, solution, "u");
```

Comment:
The simpler approach is to use the next optional argument `type_dof_data` (or something like that).